FirebaseAuth/Sources/Swift/AuthProvider/AuthProviderStrings.swift

```diff
@@ -15,12 +15,43 @@
 import Foundation
 
 /// Enumeration of the available Auth Provider IDs.
-public enum AuthProviderID: String {
-  case apple = "apple.com"
-  case email = "password"
-  case facebook = "facebook.com"
-  case gameCenter = "gc.apple.com"
-  case gitHub = "github.com"
-  case google = "google.com"
-  case phone
+public struct AuthProviderID: RawRepresentable, Equatable {
+  public var rawValue: String
+  public init(rawValue: String) {
```

Comment:
I think this should be non-public to centralize usage through the `custom(_ value: String)` API.

Reply:
Oh wait, I bet we can't do this^. The `RawRepresentable` conformance is required to allow clients to switch over the static vars, and this initializer is part of the protocol conformance and can't be non-public.

Reply:
Removed `RawRepresentable` conformance,made the `init` public and changed the `var` to a `let`.